### DISCUSSION TITLE
Pull specific file from a folder in a repo and save it at location X

### DISCUSSION THREAD

**Discussion by aebian** [community/community#22951] [New to GitHub]
<p>Greetings guys,</p>
<p>I have a question about GIT. I have a private repo here on GitHub and I would like to automate via cronjob the distrobution of a single file from it. </p>
<p>Basically I want to pull that specific file and put it into the home folder and overwrite existing file. </p>
<p>The file is located in the private repo under FolderA/.file1q2</p>
<p>Can someone guide me how this can be archieved? Is creating a empty repo really necessary? Then I would rather just select a global location for that empty report anf afterwards jut move the file via mv to the location. </p>
<p>Regards,</p>
<p>Alex</p>

**Comment by lee-dohm** [comment]
<p>There are a few different ways to achieve what you want, but yes, probably the easiest and most reliable would be to clone the repository (or pull the latest revision if you’ve already cloned it) and then copy the file you want to the location you want it to be in.</p>
<p>I hope that helps!</p>

**Comment by aebian** [comment]
<p>Hi Thanks for the reply, do you have the two other options handy? </p>
<p>Otherwise I’ll probably really need to clone the repo?</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>The only one I have a handy link to is the <a href="https://github.community/t/how-can-i-download-a-specific-folder-from-a-github-repo/278">sparse checkout technique</a>.</p>
